src/context/AppContext.tsx

```diff
@@ -0,0 +1,163 @@
+import {
+  createContext,
+  ReactNode,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { Todo } from '../types/todo';
+import useLocalStorage from '../hooks/useLocalStorage.hook';
+import { SortType } from '../enums/SortType';
+
+type AppContextContainerProps = {
+  todos: Todo[];
+  addNewTodo: (todoTitle: string) => void;
+  changeTitle: (id: number, value: string) => void;
+  clearCompleted: () => void;
+  ckickEsc: (id: number, pastTitle: string) => void;
+  editTodo: (id: number) => void;
+  makeTodosActive: () => void;
+  makeTodosCompleted: () => void;
+  removeTodo: (id: number) => void;
+  switchEdited: (id: number) => void;
+  todoCompleted: (id: number) => void;
+  sortType: SortType;
+  changeSortType: (option: SortType) => void;
+  inputRef: React.RefObject<HTMLInputElement>;
+};
+
+type Props = {
+  children: ReactNode;
+};
+
+const AppContextContainer = createContext({} as AppContextContainerProps);
+
+export const useAppContextContainer = () => {
+  return useContext(AppContextContainer);
+};
+
+export const AppContext = ({ children }: Props) => {
+  const [todos, setTodos] = useLocalStorage<Todo[]>('todos', []);
+  const [sortType, setSortType] = useState<SortType>(SortType.ALL);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const addNewTodo = (todoTitle: string) => {
+    setTodos(prev => [
+      ...prev,
+      {
+        id: +new Date(),
```

Comment:
Using +new Date() to generate a unique identifier is not a best practice, as it only provides millisecond precision and can potentially lead to collisions if multiple IDs are generated within the same millisecond. A better approach would be to use a more robust unique identifier generator, such as UUID (Universally Unique Identifier).

In JavaScript, you can use the uuid library to generate UUIDs.